marimo/_ast/compiler.py

```diff
@@ -136,7 +147,10 @@ def cell_factory(
     to generate refs and defs. If the user introduced errors to the
     signature, marimo will autofix them on save.
     """
-    function_code = textwrap.dedent(inspect.getsource(f))
+    filename = f.__code__.co_filename
+    is_script = f.__globals__["__name__"] == "__main__"
```

Comment:
I initially thought about initializing the `is_script` flag when the ScriptRuntimeContext gets initialized [here](https://github.com/marimo-team/marimo/blob/c67a2cb44f66a3c827071050e7f5e627c0829482/marimo/_runtime/app/script_runner.py#L56), and saving the value to the cell or passing it to `compile_cell()`, but it seems like the cell is compiled before the ScriptRuntimeContext is created, so I resorted to checking `__name__`.

This seems to work. I was able to run in edit mode and it did not print the name of the original script file.

Reply:
Good idea about using a dataclass! I've updated it and it looks much cleaner now.